src/providers/spatialite/qgsspatialiteprovider.cpp

```diff
@@ -4990,13 +4990,13 @@ bool QgsSpatiaLiteProvider::checkLayerType()
     // get a new alias for the subquery
     int index = 0;
     QString alias;
-    QRegExp regex;
+    QRegularExpression regex;
     do
     {
       alias = QStringLiteral( "subQuery_%1" ).arg( QString::number( index++ ) );
-      QString pattern = QStringLiteral( "(\\\"?)%1\\1" ).arg( QRegExp::escape( alias ) );
+      QString pattern = QStringLiteral( "(\\\"?)%1\\1" ).arg( QRegularExpression::escape( alias ) );
```

Comment:
escape will need ifdef as its a 5.15 thing.

Reply:
Not according to Qt's documentation and the green CI tests which compile under pre 5.15.

Reply:
Yeah maybe only the QStringView is 5.15, qt doc is unclear if this was a base thing.

Reply:
Ok confirmed -- it's in 5.11 https://doc.qt.io/archives/qt-5.11/qregularexpression.html#escape

I guess we can cleanup the ifdefs we've used elsewhere for this one then!